src/v/net/connection.cc

```diff
@@ -40,6 +40,11 @@ std::optional<ss::sstring> is_disconnect_exception(std::exception_ptr e) {
         // Happens on unclean client disconnect, typically wrapping
         // an out_of_range
         return "parse error";
+    } catch (...) {
+        // Global catch-all prevents stranded/non-handled exceptional futures.
```

Comment:
why not log something here.

Reply:
i.e.: `std::current_exception()` or just catch the exception and print it, no? `log.error()`

Reply:
Callers to this method all do already print the exception, here's an example:
```
          auto disconnect = net::is_disconnect_exception(e);
          if (disconnect) {
              vlog(
                rpclog.info,
                "Disconnected {} ({})",
                ctx->res.conn->addr,
                disconnect.value());
          } else {
              vlog(rpclog.warn, "Error dispatching method: {}", e);
          }
``

Reply:
I don't get it, in the snipped where we're logging, aren't we also already eating exceptions?  It seems like this `(...)` case is only when we're _not_ already logging it?

Reply:
In the snippet above where `disconnect` is true, we "eat" the exception in return for a descriptive string. That is what gets printed with the `disconnect` message.

Reply:
Yeah exceptions are being eaten and logged but with the missing `(...)` case the exception actually ends up escaping again into a promise where it won't be handled. I.e., the intent of this sequence is that no exception is thrown, only the existing exception is logged.